uint8 feature_no
bool is_on